support-workers/src/main/scala/com/gu/support/workers/exceptions/RetryImplicits.scala

```diff
@@ -15,8 +15,8 @@ object RetryImplicits {
       //Timeouts/connection issues and 500s
       case e @ (_: SocketTimeoutException | _: SocketException | _: WebServiceHelperError[_]) => new RetryUnlimited(message = e.getMessage, cause = throwable)
 
-      //Invalid Json
-      case e @ (_: ParsingFailure | _: MatchError) => new RetryNone(message = e.getMessage, cause = throwable)
+      //Invalid Json or http client error (4xx)
+      case e @ (_: ParsingFailure | _: MatchError | _: WebServiceClientError) => new RetryNone(message = e.getMessage, cause = throwable)
```

Comment:
We're sure that there aren't any retryable 4XX errors?

Reply:
well nothing is for sure, especially with zuora.  Having said that I have a feeling zuora errors are returned as 2xx with success=false or Success=false in the body, so I will investigate (and add tests)

Reply:
TODO- I need to check that these still work (I think they don't in this PR)
https://github.com/guardian/support-frontend/blob/ef690cebb233c7ed593840937cb3b78d2d9cc800/support-models/src/main/scala/com/gu/support/zuora/api/response/Responses.scala#L48

Reply:
I have added the tests to ZuoraErrorsSpec.scala  below and they are now passing